### PR TITLE
Feature/vue layout

### PR DESCRIPTION
**Pull request by rjmubarok**
category done

### CODE REVIEW

File: app/Http/Controllers/CategoryController.php [hunk 1]
```diff
@@ -38,28 +43,61 @@ public function store(Request $request)
...
          * active (binary default 1)
          * parent_id (nullable)
          */
-        
+        $success = false;
         $request->validate([
-            'name' => 'string',
+            'name' => 'required',
+            'name' => 'required',
+            'status' => 'required'
```
Comment (bdrosan) [inline app/Http/Controllers/CategoryController.php:58]: Why name is twice here and why status is requred?

File: app/Http/Controllers/CategoryController.php [hunk 2]
```diff
@@ -93,7 +132,47 @@ public function edit(Category $category)
      */
     public function update(Request $request, Category $category)
     {
-        //
+
+        $request->validate([
+            'name' => "required|unique:categories,name,$category->id",
```
Comment (bdrosan) [inline app/Http/Controllers/CategoryController.php:137]: What is $category->id here? Why is it different from create method?

File: app/Http/Controllers/CategoryController.php [hunk 3]
```diff
@@ -38,28 +43,61 @@ public function store(Request $request)
...
+        $file      = explode(';', $request->image);
+        $file      = explode('/', $file[0]);
+        $file_ex   = end($file);
+        $file_name = date('YmdHi') . '.' . $file_ex;
+        $success = Category::create([
+            'name'       => $request->name,
+            'slug'        => Str::slug($request->name),
+            'description'     => $request->description,
+            'status'      => $request->status,
+            'image'         =>$file_name ,
```
Comment (bdrosan) [inline app/Http/Controllers/CategoryController.php:69]: What if you have error in file upload and you won't get the $file_name?

**Comment by rjmubarok** [comment]
resolve category

**Pull request closed without merging**